normandy/recipes/models.py

```diff
@@ -121,16 +150,40 @@ def __repr__(self):
     def __str__(self):
         return self.name
 
-    def save(self, *args, **kwargs):
+    def save(self, *args, skip_last_updated=False, **kwargs):
         # Increment the revision ID and save the reversion if we've
         # changed
         if self.is_dirty(check_relationship=True):
+            dirty_fields = self.get_dirty_fields(check_relationship=True)
+
+            # Any change will invalid the signature
+            dirty_field_names = list(dirty_fields.keys())
+            if dirty_field_names != ['signature'] and self.signature is not None:
+                self.signature.delete()
```

Comment:
I'm worried about deleting the signature before the changes are saved. Would a transaction help? Or maybe just removing the signature from the recipe?